core/shared/src/main/scala/result.scala

```diff
@@ -170,6 +170,17 @@ sealed abstract class Result[+T, E <: Exception](val answer: T, val errors: Seq[
       case Unforeseen(e) => throw e
     }
 
+  /** Filter on the answer of this result. */
+  def filter(p: T => Boolean): Result[T, E] =
+    this match {
+      case Errata(_) => this
+      case Answer(b) => if(p(b)) this else Errata[T, E](Nil)
```

Comment:
On a related note, I should probably find a way to represent the state of `Errata` than a `Seq`. Perhaps separating out the `firstFailure` value from the rest?